sirepo/template/genesis.py

```diff
@@ -382,12 +414,22 @@ def _get_frame_counts(run_dir):
     return res
 
 
-def _is_text_file(path):
-    try:
-        pkio.read_text(path)
-        return True
-    except UnicodeDecodeError:
-        return False
+def _parse_maginfile(filepath):
+    if not pkio.is_pure_text(filepath):
+        raise AssertionError(f"{filepath.basename} for maginfile should be text file")
+    p = []
```

Comment:
Also add a check that the file is pure text in template.genesis validate_file().

Reply:
I thought I did. I changed the checks to `if pkio.is_pure_text(path):`

Reply:
In template.genesis validate_file() should return an error if a user tries to upload a binary maginfile. There are already checks there for partfile and distfile.